conans/model/workspace.py

```diff
@@ -1,155 +1,121 @@
 import os
-import platform
+
 from collections import OrderedDict
 
 import yaml
 
 from conans.errors import ConanException
 from conans.model.ref import ConanFileReference
-from conans.util.files import load, mkdir, save
+from conans.util.files import load, save
+from conans.model.editable_cpp_info import get_editable_abs_path
+from conans.client.graph.graph import RECIPE_EDITABLE
 
 
 class LocalPackage(object):
-    def __init__(self, base_folder, install_folder, data):
+    def __init__(self, base_folder, data, cache, ws_layout, ws_generators):
         self._base_folder = base_folder
-        self._install_folder = install_folder
-        self._conanfile_folder = data.get("folder")  # The folder with the conanfile
-        self._source_folder = data.get("source")  # By default the conanfile_folder
-        self._build_folder = data.get("build", "")
-        # package_folder can be None, then it will directly use build_folder
-        self._package_folder = data.get("package", "")
-        self._cmakedir = data.get("cmakedir")
-        includedirs = data.get("includedirs", [])  # To override include dirs, mainly for build_folder
-        self._includedirs = [includedirs] if not isinstance(includedirs, list) else includedirs
-        libdirs = data.get("libdirs", [])  # To override libdirs...
-        self._libdirs = [libdirs] if not isinstance(libdirs, list) else libdirs
-        self.conanfile = None
-
-    @property
-    def root_folder(self):
-        return os.path.abspath(os.path.join(self._base_folder, self._conanfile_folder))
-
-    @property
-    def conanfile_path(self):
-        return os.path.join(self.root_folder, "conanfile.py")
-
-    @property
-    def build_folder(self):
-        folder = self._evaluate(self._build_folder)
-        if self._install_folder:
-            pkg_folder = os.path.join(self._install_folder, self._conanfile_folder, folder)
-        else:
-            pkg_folder = os.path.join(self.root_folder, folder)
-        mkdir(pkg_folder)
-        return pkg_folder
-
-    @property
-    def package_folder(self):
-        folder = self._evaluate(self._package_folder)
-        if self._install_folder:
-            pkg_folder = os.path.join(self._install_folder, self._conanfile_folder, folder)
+        self._conanfile_folder = data.get("path")  # The folder with the conanfile
+        layout = data.get("layout")
```

Comment:
All these lines trying to resolve the layout file can be done before entering the constructor, in the `Workspace` object that is creating the `LocalPackage`s.

Also, I would unroll the members of the `data` object so this class doesn't need to know about the workspaces definition file layout.

Reply:
Leaving this for later